src/js/components/DataSearch/stories/ChangeonEnter.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+import { Data, DataSummary, DataTable, Grid, Paragraph } from 'grommet';
+
+import { DataSearch } from '../DataSearch';
+import { columns, DATA } from '../../DataTable/stories/data';
+
+export const ChangeOnEnter = () => (
+  // Uncomment <Grommet> lines when using outside of storybook
+  // <Grommet theme={...}>
+  <Grid pad="large" columns={[['medium', 'large']]} justifyContent="center">
+    <Paragraph color="text-weak">
+      Note: Results are filtered once you hit enter.
+    </Paragraph>
+    <Data data={DATA}>
+      <DataSearch updateOn="submit" />
+      <DataSummary />
+      <DataTable columns={columns} />
+    </Data>
+  </Grid>
+  // </Grommet>
+);
+
+ChangeOnEnter.args = {
+  full: true,
+};
+
+export default {
+  title: 'Data/DataSearch/ChangeOnEnter',
```

Comment:
Maybe calling this "updateOn" or "UpdateOnSubmit" might be more aligned to prop?